src/linear_algebra/dimension.lean

```diff
@@ -1355,6 +1304,65 @@ lemma module.rank_le_one_iff_top_is_principal :
   module.rank K V ≤ 1 ↔ (⊤ : submodule K V).is_principal :=
 by rw [← submodule.rank_le_one_iff_is_principal, dim_top]
 
+end division_ring
+
+end module
+
+/-! ### The rank of a linear map -/
+
+namespace linear_map
+
+section ring
+variables [ring K] [add_comm_group V] [module K V] [add_comm_group V₁] [module K V₁]
+variables [add_comm_group V'] [module K V']
+
+/-- `rank f` is the rank of a `linear_map` `f`, defined as the dimension of `f.range`. -/
+def rank (f : V →ₗ[K] V') : cardinal := module.rank K f.range
```

Comment:
Should we even make this protected? I suppose not because it used to be in the root namespace and didn't cause too many issues in my recollection.
```suggestion
protected def rank (f : V →ₗ[K] V') : cardinal := module.rank K f.range
```

Reply:
I think making it protected would be fine, but it would make more downstream work in this PR and like you say it's probably harmless unprotected.